sat/cli/jobstat/main.py

```diff
@@ -0,0 +1,79 @@
+# MIT License
+#
+# (C) Copyright 2022 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+"""
+The main entry point for the jobstat subcommand.
+"""
+import logging
+import requests
+from sat.cli.jobstat import parser
+from sat.report import Report
+from sat.config import get_config_value
+
+
+URL = "http://localhost:1999/all"
```

Comment:
jobstat is running inside a pod. So, changed it to https://api-gw-service-nmn.local/apis/statechecker/jobstat/all

Reply:
Sounds good, that is much closer to what we would want here. SAT has a configuration file that lets you configure the API gateway hostname, so I would suggest we use that instead of hard-coding `api-gw-service-nmn.local` here.

E.g. something like:
```
URL = f'https://{get_config_value("api_gateway.host")/apis/statechecker/jobstat/all'
```

Follow-up question, is there or will there be any authentication on this endpoint? If so (and possibly even if not) it might make sense to create a subclass of `APIGatewayClient` ([source](https://github.com/Cray-HPE/sat/blob/integration/sat/apiclient/gateway.py#L69)) - and put some of the query logic into this class instead of in the `main` function. I think this would definitely make unit testing easier.

Then it would be a little more consistent with some of the other commands. Here are some examples of these type of classes:
* BOS: https://github.com/Cray-HPE/sat/blob/integration/sat/apiclient/bos.py
* CAPMC: https://github.com/Cray-HPE/sat/blob/integration/sat/apiclient/capmc.py
* CFS: https://github.com/Cray-HPE/sat/blob/integration/sat/apiclient/cfs.py

Doing that would end up removing the need to define `URL` at all above, it would also be a little bit of a refactor of the current implementation.

Reply:
Replaced this with code
`        connect = APIGatewayClient()
        return connect.get(['statechecker', 'jobstat', 'all']).json()`